backend/inspirehep/assign/tasks.py

```diff
@@ -0,0 +1,141 @@
+import structlog
+from celery import shared_task
+from inspire_schemas.builders import LiteratureBuilder
+from invenio_db import db
+from invenio_pidstore.errors import PIDDoesNotExistError
+from jsonschema import ValidationError
+from sqlalchemy.exc import (
+    DisconnectionError,
+    OperationalError,
+    ResourceClosedError,
+    TimeoutError,
+    UnboundExecutionError,
+)
+from sqlalchemy.orm.exc import NoResultFound, StaleDataError
+
+from inspirehep.records.api import ConferencesRecord, LiteratureRecord
+from inspirehep.records.errors import MissingArgumentError
+
+LOGGER = structlog.getLogger()
+
+
+@shared_task(
+    ignore_results=False,
+    queue="assign",
+    acks_late=True,
+    retry_backoff=2,
+    retry_kwargs={"max_retries": 6},
+    autoretry_for=(
+        NoResultFound,
+        StaleDataError,
+        DisconnectionError,
+        TimeoutError,
+        UnboundExecutionError,
+        ResourceClosedError,
+        OperationalError,
+    ),
+)
+def assign_paper_to_conference(literature_recids, conference_recid):
+    try:
+        conference = ConferencesRecord.get_record_by_pid_value(conference_recid)
+    except PIDDoesNotExistError:
+        LOGGER.exception(
+            "Cannot assign papers to conference. Conference does not exist.",
+            conference_recid=conference_recid,
+        )
+    conference_ref = conference.get("self")
+    cnum = conference.get("cnum")
+    for recid in literature_recids:
+        try:
+            record = LiteratureRecord.get_record_by_pid_value(recid)
+        except PIDDoesNotExistError:
+            LOGGER.error(
+                "Cannot assign record to conference. Record does not exist.",
+                literature_recid=recid,
+                conference_recid=conference_recid,
+            )
+            continue
+        with db.session.begin_nested():
+            try:
+                updated_data = assign_conference(record, conference_ref, cnum)
+                record.update(updated_data)
+            except ValidationError as err:
+                LOGGER.exception(
+                    "Cannot assign conference to paper.",
+                    recid=recid,
+                    cnum=cnum,
+                    conference_recid=conference_recid,
+                    err=err,
+                )
+            except MissingArgumentError:
+                LOGGER.error(
+                    "CNUM and conference $ref are required.",
+                    cnum=cnum,
+                    conference_ref=conference_ref,
+                    record_recid=recid,
+                )
+    db.session.commit()
+
+
+def assign_conference(record, conference_ref, cnum):
+    builder = LiteratureBuilder(record=record)
+    if not cnum:
+        raise MissingArgumentError("cnum is required.")
+    if not conference_ref:
+        raise MissingArgumentError("$ref is required.")
+
+    if not {"proceedings", "conference paper"}.intersection(
```

Comment:
Why is that? For example if `document_type` is `article` why you are changing it?

Reply:
> So finally we always pass recids to the task (even if all the records were selected)?
Yes.

Reply:
I'm not changing this, I'm adding another document_type. This requirement was added when I was working on the task https://github.com/inspirehep/inspirehep/issues/1768#issuecomment-800209835